etl/steps/data/grapher/wb/2024-07-29/income_groups.py

```diff
@@ -1,22 +1,43 @@
 """Load a garden dataset and create a grapher dataset."""
 
-from etl.helpers import PathFinder, create_dataset
+from etl.helpers import PathFinder
 
 # Get paths and naming conventions for current step.
 paths = PathFinder(__file__)
 
 
-def run(dest_dir: str) -> None:
+def run() -> None:
     #
     # Load inputs.
     #
-    # Load garden dataset and read its table on historical income classifications.
+    # Load garden dataset.
     ds_garden = paths.load_dataset("income_groups")
-    tb_garden = ds_garden["income_groups"]
+
+    # Read table of income groups (a dynamic classification that changes over the years).
+    tb_dynamic = ds_garden.read("income_groups")
+
+    # Read table of latest income groups (a static classification).
+    tb_static = ds_garden.read("income_groups_latest")
+
+    #
+    # Process data.
+    #
+    # Combine both tables.
+    tb = tb_dynamic.merge(tb_static, on=["country"], how="outer", suffixes=("", "_latest"))
+
```

Comment:
is this merge needed as a way to "hack" the country values for all years?

Reply:
I don't think we need **all** years defined, but we may need at least **one** year (I haven't checked if it works without a "year" column). Do you see any issue with the current approach?

Reply:
For more context, the `tb_static` table contains just two columns, namely `country`, and `classification`. We could create a separate dataset for it, but I suppose that may be more inconvenient than having it as part of the dynamic one, and it may not work as "color" if it doesn't have a "year" column (I haven't checked, and I don't have a strong opinion).

Reply:
OK, I do see an issue with my approach, which is that, if someone looks at the "Table", it may be confusing to see multiple years from 1987 onwards (with the same classification). So I'll create the two separate tables.

Reply:
I've just changed the code to have two separate tables, and remapped the indicators in the charts. And I've just noticed that this is not ideal either: if you see the Table tab of, e.g. [this chart](http://staging-site-data-income-group-grapher/admin/charts/3776/edit) you don't see any income classification (since the latest point in the axes is 2021, which is smaller than 2023, the latest point for which the income classification has data).

To be able to see the classification for all years, we'd need to add all possible country-years with the latest classification.

Anyway, I'll keep it the way you suggested, and we can rethink it in the future.